File: Includes/TableClasses/Category.php
<?php
class Category
{

    private $Table = "categories";
    public $id, $parent_id, $level, $name, $order_level, $commision_rate, $banner, $icon, $featured, $top, $digital, $slug, $meta_title, $meta_description, $created_at, $updated_at;
    private $conn;

    private $imagePathRoot  = "https://d2t03bblpoql2z.cloudfront.net/";


    public function __construct($con, $id)
    {
        $this->conn = $con;
        $this->id = $id;

        $stmt = $this->conn->prepare("SELECT `id`, `parent_id`, `level`, `name`, `order_level`, `commision_rate`, `banner`, `icon`, `featured`, `top`, `digital`, `slug`, `meta_title`, `meta_description`, `created_at`,  `updated_at` FROM " . $this->Table . " WHERE id = ? ORDER BY featured DESC");
        $stmt->bind_param("i", $this->id);
        $stmt->execute();
        $stmt->bind_result($this->id, $this->parent_id, $this->level, $this->name, $this->order_level, $this->commission_rate, $this->banner, $this->icon, $this->featured, $this->top, $this->digital, $this->slug, $this->meta_title, $this->meta_description, $this->created_at, $this->updated_at);

        while ($stmt->fetch()) {
            $this->id = $this->id;
            $this->parent_id = $this->parent_id;
            $this->level =  $this->level;
            $this->name =  $this->name;
            $this->order_level =  $this->order_level;
            $this->commision_rate = $this->commision_rate;
            $this->banner = $this->banner;
            $this->icon = $this->icon;
            $this->featured = $this->featured;
            $this->top = $this->top;
            $this->digital = $this->digital;
            $this->slug =  $this->slug;
            $this->meta_title = $this->meta_title;
            $this->meta_description = $this->meta_description;
            $this->created_at =  $this->created_at;
            $this->updated_at = $this->updated_at;
        }
    }




    function getCategoryProducts()
    {
        $categoryids = array();
        $categoryMenuItems = array();


        $category_stmt = "SELECT id FROM products  WHERE published = 1 AND category_id = ". $this->id. " ORDER BY num_of_sale LIMIT 6";
		$menu_type_id_result = mysqli_query($this->conn, $category_stmt);

		while ($row = mysqli_fetch_array($menu_type_id_result)) {

			array_push($categoryids, $row);
		}

    
		foreach ($categoryids as $row) {
			$product = new Product($this->conn, intval($row['id']));
			$temp = array();
			$temp['id'] = $product->getId();
			$temp['name'] = $product->getName();
			$temp['category_id'] = $product->getCategory_id();
            $temp['photos'] = $product->getPhotos();
			$temp['thumbnail_img'] = $product->getThumbnail_img();
			$temp['unit_price'] = $product->getUnit_price();
            $temp['discount'] = $product->getDiscount();
            $temp['purchase_price'] = $product->getPurchase_price();
			$temp['meta_title'] = $product->getMeta_title();
			$temp['meta_description'] = $product->getMeta_description();
            $temp['meta_img'] = $product->getMeta_img();
            $temp['min_qtn'] = $product->getMin_qty();
            $temp['published'] = $product->getPublished();
	
			array_push($categoryMenuItems, $temp);
		}


        return $categoryMenuItems;
    }


    /**
     * Get the value of id
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Get the value of level
     */
    public function getLevel()
    {
        return $this->level;
    }

    /**
     * Get the value of parent_id
     */
    public function getParent_id()
    {
        return $this->parent_id;
    }

    /**
     * Get the value of name
     */
    public function getName()
    {
        return $this->name;
    }

    /**
     * Get the value of order_level
     */
    public function getOrder_level()
    {
        return $this->order_level;
    }

    /**
     * Get the value of commission_rate
     */
    public function getCommission_rate()
    {
        return $this->commission_rate;
    }

    /**
     * Get the value of banner
     */
    public function getBanner()
    {
        $upload = new Upload($this->conn, $this->banner);
        $filename = $this->imagePathRoot . $upload->getFile_name();

        return $filename;
    }

    /**
     * Get the value of icon
     */
    public function getIcon()
    {
        return $this->icon;
    }

    /**
     * Get the value of featured
     */
    public function getFeatured()
    {
        return $this->featured;
    }

    /**
     * Get the value of top
     */
    public function getTop()
    {
        return $this->top;
    }

    /**
     * Get the value of digital
     */
    public function getDigital()
    {
        return $this->digital;
    }

    /**
     * Get the value of slug
     */
    public function getSlug()
    {
        return $this->slug;
    }

    /**
     * Get the value of meta_title
     */
    public function getMeta_title()
    {
        return $this->meta_title;
    }

    /**
     * Get the value of meta_description
     */
    public function getMeta_description()
    {
        return $this->meta_description;
    }

    /**
     * Get the value of created_at
     */
    public function getCreated_at()
    {
        $phpdate = strtotime($this->created_at);
        $mysqldate = date('d M Y', $phpdate);
        return $mysqldate;
    }

    /**
     * Get the value of updated_at
     */
    public function getUpdated_at()
    {

        $phpdate = strtotime($this->updated_at);
        $mysqldate = date('d M Y', $phpdate);
        return $mysqldate;
    }

}

File: Category.java

package com.example;

import java.util.List;
import javax.annotation.Generated;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Generated("jsonschema2pojo")
public class Category {

    @SerializedName("previousSearches")
    @Expose
    private PreviousSearches previousSearches;
    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("parent_id")
    @Expose
    private Integer parentId;
    @SerializedName("level")
    @Expose
    private Integer level;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("order_level")
    @Expose
    private Integer orderLevel;
    @SerializedName("commision_rate")
    @Expose
    private Integer commisionRate;
    @SerializedName("banner")
    @Expose
    private String banner;
    @SerializedName("icon")
    @Expose
    private String icon;
    @SerializedName("featured")
    @Expose
    private Integer featured;
    @SerializedName("top")
    @Expose
    private Integer top;
    @SerializedName("digital")
    @Expose
    private Integer digital;
    @SerializedName("slug")
    @Expose
    private String slug;
    @SerializedName("meta_title")
    @Expose
    private Object metaTitle;
    @SerializedName("meta_description")
    @Expose
    private Object metaDescription;
    @SerializedName("created_at")
    @Expose
    private String createdAt;
    @SerializedName("updated_at")
    @Expose
    private String updatedAt;
    @SerializedName("products")
    @Expose
    private List<Product> products = null;

    public PreviousSearches getPreviousSearches() {
        return previousSearches;
    }

    public void setPreviousSearches(PreviousSearches previousSearches) {
        this.previousSearches = previousSearches;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getParentId() {
        return parentId;
    }

    public void setParentId(Integer parentId) {
        this.parentId = parentId;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getOrderLevel() {
        return orderLevel;
    }

    public void setOrderLevel(Integer orderLevel) {
        this.orderLevel = orderLevel;
    }

    public Integer getCommisionRate() {
        return commisionRate;
    }

    public void setCommisionRate(Integer commisionRate) {
        this.commisionRate = commisionRate;
    }

    public String getBanner() {
        return banner;
    }

    public void setBanner(String banner) {
        this.banner = banner;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public Integer getFeatured() {
        return featured;
    }

    public void setFeatured(Integer featured) {
        this.featured = featured;
    }

    public Integer getTop() {
        return top;
    }

    public void setTop(Integer top) {
        this.top = top;
    }

    public Integer getDigital() {
        return digital;
    }

    public void setDigital(Integer digital) {
        this.digital = digital;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(String slug) {
        this.slug = slug;
    }

    public Object getMetaTitle() {
        return metaTitle;
    }

    public void setMetaTitle(Object metaTitle) {
        this.metaTitle = metaTitle;
    }

    public Object getMetaDescription() {
        return metaDescription;
    }

    public void setMetaDescription(Object metaDescription) {
        this.metaDescription = metaDescription;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }

    public List<Product> getProducts() {
        return products;
    }

    public void setProducts(List<Product> products) {
        this.products = products;
    }

}
